apps/vaporgui/VDoubleLineEdit.cpp

```diff
@@ -0,0 +1,79 @@
+#include <string>
+#include <sstream>
+#include <iomanip>
+
+#include <QString>
+
+#include "VDoubleLineEdit.h"
+
+VDoubleLineEdit::VDoubleLineEdit( double value, bool useMenu ) : 
+    VNumericLineEdit(),
+    _value( value )
+{
+    // Disconnect the QLineEdit's interaction with VStringLineEdit::_valueChanged(),
+    // and reconnect it to VDoubleLineEdit::_valueChanged()
+    disconnect( _lineEdit, SIGNAL( editingFinished() ),
+        this, SLOT( _valueChanged() ) );
+    connect( _lineEdit, SIGNAL( editingFinished() ),
+        this, SLOT( _valueChanged() ) );
+
+    std::string formattedNumber = _formatValue( _value );
+    SetValueString( formattedNumber );
+}
+
+void VDoubleLineEdit::SetValueDouble( double value ) {
+    std::string formattedNumber = _formatValue( value );
+
+    try {
+        _value = std::stod(formattedNumber);
+    } catch (const std::invalid_argument&) {
+        return;
+    } catch (const std::out_of_range&) {
+        return;
+    }
+
+    SetValueString( formattedNumber );
+}
+
+double VDoubleLineEdit::GetValueDouble() const {
+    return _value;
+}
+
+void VDoubleLineEdit::_valueChanged() {
+    std::string str = _getText();
+
+    double value;
+    try {
+        value = std::stod( str );
+        SetValueDouble( value );
+        emit ValueChanged( _value );
+    } catch (const std::invalid_argument&) {
+        SetValueDouble( _value );
+    } catch (const std::out_of_range&) {
+        SetValueDouble( _value );
+    }
+/*    bool legalConversion;
+    QString str = _lineEdit->text();
+    double value = str.toDouble( &legalConversion );
+
+    if (legalConversion) {
+        SetValueDouble( value );
+        emit ValueChanged( _value );
+    }
+    else {
+        SetValueDouble( _value );
+    }*/
+}
+
+std::string VDoubleLineEdit::_formatValue( double value ) {
```

Comment:
VIntLineEdit::_formatValue and VIntDoubleEdit::_formatValue are identical and VIntLineEdit::_formatValue casts the int to a double anyway so you could put this in the parent class.

Reply:
There is actually a subtle difference.

The double version just operates on doubles, plain and simple.

However the integer version casts its value to a double _only if_ scientific notation is enabled.